airflow/jobs/scheduler_job.py

```diff
@@ -1337,24 +1337,40 @@ def _process_executor_events(self, simple_dag_bag, session=None):
 
                 # TODO: should we fail RUNNING as well, as we do in Backfills?
                 if ti.try_number == try_number and ti.state == State.QUEUED:
-                    msg = ("Executor reports task instance {} finished ({}) "
-                           "although the task says its {}. Was the task "
-                           "killed externally?".format(ti, state, ti.state))
-                    Stats.incr('scheduler.tasks.killed_externally')
-                    self.log.error(msg)
-                    try:
-                        simple_dag = simple_dag_bag.get_dag(dag_id)
-                        dagbag = models.DagBag(simple_dag.full_filepath)
-                        dag = dagbag.get_dag(dag_id)
-                        ti.task = dag.get_task(task_id)
-                        ti.handle_failure(msg)
-                    except Exception:
-                        self.log.error("Cannot load the dag bag to handle failure for %s"
-                                       ". Setting task to FAILED without callbacks or "
-                                       "retries. Do you have enough resources?", ti)
-                        ti.state = State.FAILED
-                        session.merge(ti)
-                        session.commit()
+                    task_reschedules = TaskReschedule.find_for_task_instance(ti)
```

Comment:
I feel like we could have a method that finds task reschedules without getting the whole list of them. E.G. an exists() function

Also, if I did externally kill the task during reschedule phase, or whatever is causing the disagreement, this would treat it like a reschedule, correct? Is there a way we can compensate for that? It sounds like the non-hacky fix is to have the ordinary retry number, and a reschedule number as well. and we can check that that number incremented. But that might be more work at the moment that we should put into this fix. I guess I'm still a little confused on what causes to hit this state under normal conditions.